tests/openassetio-core/trait/TraitBaseTest.cpp

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#include <openassetio/trait/TraitBase.hpp>
+#include <openassetio/typedefs.hpp>
+
+#include <catch2/catch.hpp>
+#include "openassetio/TraitsData.hpp"
+
+namespace {
+
+using openassetio::Str;
+using openassetio::TraitsData;
+using openassetio::TraitsDataPtr;
+using openassetio::trait::TraitBase;
+using openassetio::trait::TraitId;
+using openassetio::trait::TraitPropertyStatus;
+using openassetio::trait::property::Key;
+
+// TraitBase can't be tested directly, so we derive a test trait.
+struct TestTrait : TraitBase<TestTrait> {
```

Comment:
[consider][minor] An alternative would be to put `TestTrait` in the `openassetio::trait` namespace, so you wouldn't need all those `using` aliases.  Note that C++17 supports a concise namespace syntax, e.g.
```
namespace openassetio::trait {
struct TestTrait  {
```

Reply:
I suppose - as this is within the project, this is less heinous?

Reply:
Yeah I suppose. The compiler should give you some confidence that you're not redeclaring symbols, though I guess it's still possible. You could add another child namespace to avoid any chance of ambiguity, e.g.
```
namespace openassetio::trait::test {
struct TestTrait  {
```

:thinking: on balance, it's probably better the way you've already done it. 

I would perhaps opt for more like `namespace trait = openassetio::trait` to hoist the namespace rather than the leaf types, but minor, that's just a personal style.

I'll resolve this.

Reply:
Updated to more compact traits alias [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/373/commits/5b227b86cbd217567383f156948a366d36eecad6).